modules/shared/src/main/scala/org/tessellation/schema/swap.scala

```diff
@@ -0,0 +1,78 @@
+package org.tessellation.schema
+
+import org.tessellation.schema.address.Address
+import org.tessellation.security.hash.Hash
+
+import derevo.cats.{order, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import eu.timepit.refined.types.numeric.PosLong
+import io.estatico.newtype.macros.newtype
+
+object swap {
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class SwapAmount(value: PosLong)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class SwapSalt(value: Long)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class AllowSpendFee(value: PosLong)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class CurrencyId(value: Address)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class AllowSpendReference(value: Hash) // exact type TBD (e.g. lastRef?)
```

Comment:
is it a `TODO`? IMO you should create a ticket for that as it can easily get lost. otherwise, is the PR ready to be merged?

Reply:
I think we should just use the same fields as transaction reference here.

Reply:
changed